### PR TITLE
LWDEV-10401 Add health check of elements count in inner data queues

### PR DESCRIPTION
**Pull request by rocketmannot**

### CODE REVIEW

File: src/main/kotlin/com/lykke/matching/engine/config/spring/QueueConfig.kt [hunk 1]
```diff
@@ -120,36 +120,90 @@ open class QueueConfig {
...
+
+    @Bean
+    @DataQueue
+    open fun marketOrderWithTradesLogQueue(): BlockingQueue<com.lykke.matching.engine.logging.MessageWrapper> {
+        return LinkedBlockingQueue()
+    }
+
+    @Bean
+    @DataQueue
+    open fun orderBookLogQueue(): BlockingQueue<com.lykke.matching.engine.logging.MessageWrapper> {
```
Comment (barsukovskii) [inline src/main/kotlin/com/lykke/matching/engine/config/spring/QueueConfig.kt]: We don't log orderbooks to db, so this queue is not needed

File: src/main/kotlin/com/lykke/matching/engine/config/spring/QueueConfig.kt [hunk 2]
```diff
@@ -120,36 +120,90 @@ open class QueueConfig {
...
+    @Bean
+    @DataQueue
+    open fun orderBookQueue(): BlockingQueue<OrderBook> {
+        return LinkedBlockingQueue<OrderBook>()
+    }
+    //</editor-fold>
+
+    //<editor-fold desc="Etc queues">
+    @Bean
+    open fun dbTransferOperationQueue(): BlockingQueue<TransferOperation> {
```
Comment (barsukovskii) [inline src/main/kotlin/com/lykke/matching/engine/config/spring/QueueConfig.kt]: maybe we also should add this queue to monitoring ?

File: src/main/kotlin/com/lykke/matching/engine/config/spring/AppConfiguration.kt [hunk 3]
```diff
@@ -47,14 +47,23 @@ open class AppConfiguration {
     }
 
     @Bean
-    open fun rabbitQueueSizeChecker(@RabbitQueue namesToInputQueues: Map<String, BlockingQueue<*>>): QueueSizeHealthChecker {
+    open fun rabbitQueueSizeChecker(@RabbitQueue  namesToInputQueues: Map<String, BlockingQueue<*>>): QueueSizeHealthChecker {
```
[the remark this answers is not in the document]
Comment (rocketmannot) [reply src/main/kotlin/com/lykke/matching/engine/config/spring/AppConfiguration.kt]: fixed

File: src/main/kotlin/com/lykke/matching/engine/config/spring/QueueConfig.kt [hunk 4]
```diff
@@ -120,36 +120,90 @@ open class QueueConfig {
     }
     //</editor-fold>
 
+    //<editor-fold desc="Data queues">
+    @Bean
+    @DataQueue
+    open fun balanceUpdatesLogQueue(): BlockingQueue<com.lykke.matching.engine.logging.MessageWrapper> {
```
[the remark this answers is not in the document]
Comment (rocketmannot) [reply src/main/kotlin/com/lykke/matching/engine/config/spring/QueueConfig.kt]: fixed

File: src/main/kotlin/com/lykke/matching/engine/config/spring/QueueConfig.kt (hunk 2 above)
Comment (rocketmannot) [reply src/main/kotlin/com/lykke/matching/engine/config/spring/QueueConfig.kt]: added

**Pull request merged**